.github/workflows/deploy.yml

```diff
@@ -0,0 +1,71 @@
+name: Build and Deploy to S3
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - development
+      - base
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    environment:
+      name: development ##Replace with actual environment after testing
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```

Comment:
I'm sure you've looked at this but can you explain to me why this is necessary?

Reply:
So, GItHub recommends checking out the code anytime the workflow will interact with the repository's code. "This is an action that checks out your repository onto the runner, allowing you to run scripts or other actions against your code (such as build and test tools). You should use the checkout action any time your workflow will use the repository's code"

Additionally, I believe that unless it's a PR action, GitHub will use the default branch, instead of the working branch.

https://stackoverflow.com/questions/76938677/whats-the-purpose-of-actions-checkoutv3-when-the-repository-is-already-checke

Reply:
By default, the action will check-out the SHA for that workflow’s event (such as push and pull_request). Otherwise, it uses the default branch.

Reply:
Got it, that makes sense.